src/rgw/rgw_gc.cc

```diff
@@ -37,7 +37,29 @@ void RGWGC::initialize(CephContext *_cct, RGWRados *_store) {
     char buf[32];
     snprintf(buf, 32, ".%d", i);
     obj_names[i].append(buf);
+
+    std::list<cls_rgw_gc_obj_info> entries;
+    string marker, next_marker;
+    bool truncated = false;
+    auto ret = cls_rgw_gc_list(store->gc_pool_ctx, obj_names[i], marker, 1,
+			  false, entries, &truncated, next_marker);
+    auto it = transitioned_objects_cache.begin() + i;
+    if (ret == -ENOENT || entries.size() == 0) {
+      transitioned_objects_cache.insert(it, true);
+    } else { // for failure cases also?
+      transitioned_objects_cache.insert(it, false);
+    }
+    ldpp_dout(this, 20) << "RGWGC::initialize object name = " << obj_names[i] << ", transitioned = " << transitioned_objects_cache[i] << dendl;
+
+    // Create GC queues
+    ldpp_dout(this, 20) << "RGWGC::initialize initing gc queue with name = " << obj_names[i] << dendl;
+    librados::ObjectWriteOperation op;
+    op.create(true);
+    uint64_t queue_size = 1048576, num_urgent_data_entries = 50;
+    cls_rgw_gc_queue_init(op, obj_names[i], queue_size, num_urgent_data_entries);
```

Comment:
i think we want the queue size (and maybe urgent entries too?) to configurable. that will be useful for testing the ENOSPC handling, because you can set an artificially low queue size

i feel like the default size should be much higher than 1M - i thought we were looking at limits on the osd in the hundred megs range (cc @adamemerson)

Reply:
128MiB by default, can be increased to 4GiB in ceph.conf.

Reply:
okay, Adam pointed me at `osd_max_object_size` which defaults to 128M. if we add a config option for the gc queue size, it should mention that it should be no larger than that osd config value

Reply:
Yeah ok, will add the configs